llama.cpp

```diff
@@ -2955,18 +3037,18 @@ struct llama_model_loader {
     }
 
     ~llama_model_loader() {
-        if (ctx_gguf) {
-            gguf_free(ctx_gguf);
+        if (meta) {
+            gguf_free(meta);
         }
-        if (ctx_meta) {
-            ggml_free(ctx_meta);
+        for (auto & ctx : contexts) {
```

Comment:
The contexts are pointers, no need to iterate by reference.
```suggestion
        for (auto * ctx : contexts) {
```